docs/dg/dev/guidelines/project-development-guidelines.md

```diff
@@ -23,42 +23,44 @@ There are different strategies developers can use to develop projects. Before st
+
 ## Apply coding guidelines
-Starting from the first day of development, apply the [coding guidelines](/docs/dg/dev/guidelines/coding-guidelines/coding-guidelines.html).
+Starting from the first day of development, apply the [coding guidelines](/docs/dg/dev/guidelines/coding-guidelines/coding-guidelines.html). 
+Pay attention to an [architecture convention](/docs/dg/dev/architecture/architectural-convention.html) page as in addition it provides a set of rules and recommendations that applicable specifically for project development.
 
 ## Use custom namespaces
-Instead of using the `Pyz` namespace, create your own namespace for project development. For example, use the project name as a namespace.
+As a recommendation, instead of using the `Pyz` namespace, you can use your own namespace for project development. For example, use the project name as a namespace. But be advised that most of existing examples and documentation use the `Pyz` namespace.
 
 ## Use custom names
 Use custom names for everything that is added on the project level, like the following:
 - Transfers
 - Transfer properties
-- DB tables
+- DB tables (using project namespace as a prefix is recommended)
 - DB table fields,
 - Modules
 - Classes
-- Interfaces
-- Traits
-- Properties
-- Contants
 
-For example, customize the names by adding the project name.
+For example, customize the names by adding the project name. This will help to avoid conflicts with the Spryker core and other projects.
 
 ## Avoid using, extending, and overriding Private API
-Instead of using, extending, and overriding [Private API](/docs/dg/dev/architecture/module-api/declaration-of-module-apis-public-and-private.html), send a request about the missing endpoints to your Spryker account manager. Spryker offers extension points that allow you to extend via the Public API and helps you to customize the application. We recommend to use the extension points instead of overriding Private API.
+Instead of using, extending, and overriding [Private API](/docs/dg/dev/architecture/module-api/declaration-of-module-apis-public-and-private.html) (everything that is not a Public API), send a request about the missing endpoints to your Spryker account manager. Spryker offers extension points that allow you to extend via the Public API and helps you to customize the application. We recommend to use the extension points instead of overriding Private API.
 
-## Avoid using unsupported types
-Avoid using unsupported types as constructor arguments of a plugin.
+Extending Private API is still possible, but core development teams are not considering changing those classes as any BC break as long as a Public API is not affected. So you can get a critical error with your next minor update.
```

Comment:
```suggestion
Extending Private API is still possible, but Spryker development team can change them without a notice in the minor change, since Spryker BC break policy is only considering a Public API.
```